src/main.c

```diff
@@ -235,6 +243,47 @@ main_loop(void)
 
 	config = config_get_config();
 
+	// Check for libmicrohttp version at runtime, ie actual installed version
+	int major = 0;
+	int minor = 0;
+	int patch = 0;
+	const char *version = MHD_get_version();
+
+	debug(LOG_NOTICE, "MHD version is %s", version);
+
+	if (sscanf(version, "%d.%d.%d", &major, &minor, &patch) == 3) {
+
+		if (major < MIN_MHD_MAJOR) {
```

Comment:
Please shorten the code and put it into it's own method.

Reply:
Mmm, yes, a bit repetitive.

Reply:
Yes, that is better. But please but the logic in a separate method. Either a method to get the MHD version or a method to check the version:
```
// return libmicrohttpd version in hex format (same as MHD_VERSION)
unsigned get_microhttpdversion() {
  unsigned major = 0;
  unsigned minor = 0;
  unsigned patch = 0;
  const char *version = MHD_get_version();
  if (sscanf(version, "%x.%x.%x", &major, &minor, &patch) == 3) {
    return (major << 24) | (minor << 16) | (patch << 8);
  } else {
     return 0;
  }
}

int
main(int argc, char *argv[])
{
  unsigned required_version = 0x00096900;
  unsigned found_version = get_microhttpdversion();
  if (found_version < required_version) {
    printf("too old\n");
  }
  return 0;
}
```

Reply:
Well, it is either:
 * split up into major, minor patch and do up to 3 tests
 * split up into major, minor patch, convert to hex format then do 1 test

Also, we only need to check once on startup so there is little point in using a function.

Yours is smaller than mine (but mine logs the required version) ;-)

As I am very short of time I would like to leave it as it is.

Note: I kept the original check for MHD_VERSION at 0x00095100 as 0.9.51 is the minimum required microhttpd.h for NDS to be compiled with. As we know, this is not related in any way to the MHD version that may be available at runtime.
0.9.51 is good enough if using only an external FAS, but can be problematic with form data if using MHD to serve the client page(s).

Reply:
My point was about to put the code to acquire the version into a separate method. This way it does not pollute the namespace of the main() method.

Reply:
Having looked up "namespace pollution", I see what you mean.
main_loop() as it now stands is badly named, it should be more like "setup_then_start_main_loop".

I think we should do some more general housekeeping in the near future, as the subject of one or more new PRs